### PR TITLE
KYLIN-5280 Introduce memcached as a distributed cache for queries for Kylin5

### PR DESCRIPTION
**Pull request by zznlime**
## Branch to commit
- [ ] Branch **kylin3** for v2.x to v3.x
- [ ] Branch **kylin4** for v4.x
- [x] Branch **kylin5** for v5.x

## Types of changes

What types of changes does your code introduce to Kylin?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have created an issue on [Kylin's jira](https://issues.apache.org/jira/browse/KYLIN), and have described the bug/feature there in detail
- [x] Commit messages in my PR start with the related jira ID, like "KYLIN-5280 Introduce memcached as a distributed cache for queries for Kylin5"
- [x] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged

### CODE REVIEW

**Comment by hit-lacus** [comment]
retest this please

File: src/common-service/src/main/java/org/apache/kylin/rest/cache/redis/RedisCache.java [hunk 1]
```diff
@@ -67,7 +68,7 @@ public class RedisCache implements KylinCache {
 
     private static final String NX = "NX";
     private static final String XX = "XX";
-    private static final String PREFIX = "Kyligence-";
+    private static final String PREFIX = "Kylin-";
```
Comment (hit-lacus) [inline src/common-service/src/main/java/org/apache/kylin/rest/cache/redis/RedisCache.java:71]: I will check if this change will introduce some incompatible issue.

Comment (hit-lacus) [reply src/common-service/src/main/java/org/apache/kylin/rest/cache/redis/RedisCache.java:71]: I think it is ok for me.

**Comment by hit-lacus** [comment]
This patch looks good to me. Could you please fix the code smell issues? Try to make these issues to zero, if you find some issue are not necessary to fix, you may need this guide (https://9to5answer.com/how-can-we-ignore-some-sonarqube-rules-in-java) .

image

Thanks, @zznlime , I will merge it after code smell issue are done.

**Comment by hit-lacus** [comment]
After this patch be merged, could you please kindly add doc for memcached cache in the page (https://github.com/apache/kylin/blob/doc5.0/website/docs/configuration/query_cache.md) ?

**Comment by zznlime** [comment]
will close this pull request since a new pr for this feature has been opened #2096

**Pull request closed without merging**